panic: Some of the data has been pending for too long

When running minimal (commit 1b541abee225f3cd3f3f20bf6bbfe4d236195c97) sync for a while, getting this stack trace:
```
SYNC RECEIPTS (90a4c4c617ae6431929dc83d3b720c7c0bbe754f38821042ec528d983be39e95a0f49790692f72fc34acfcd7b5d2662691f631e8eb9e33fd888675605b5cee28): 47
deliver receipts 392: 50
deliver receipts 393: 54
SYNC BODIES (90a4c4c617ae6431929dc83d3b720c7c0bbe754f38821042ec528d983be39e95a0f49790692f72fc34acfcd7b5d2662691f631e8eb9e33fd888675605b5cee28): 52
SYNC HEADERS (867c7bfd6df41ad8537eced23c8c48a4b474151156be0bbd9920f391ae0bc38ef6b2319b4f1afd96a8ab1e3bc5a8e9f340cf94a6f6ef8b9726436c013f7d39bb): 165301 190
deliver bodies 392: 53
block 101461: Completed Pending Completed
block 101651: Completed Completed Pending
block 101841: Completed Completed Completed
block 102031: Completed Completed Completed
block 102221: Completed Completed Completed
block 102411: Completed Completed Completed
block 102601: Completed Completed Completed
block 102791: Completed Completed Completed
block 102981: Completed Completed Completed
block 103171: Completed Completed Completed
block 103361: Completed Completed Completed
block 103551: Completed Completed Completed
block 103741: Completed Completed Completed
block 103931: Completed Completed Completed
block 104121: Completed Completed Completed
block 104311: Completed Completed Completed
block 104501: Completed Completed Completed
block 104691: Completed Completed Completed
block 104881: Completed Completed Completed
block 105071: Completed Completed Completed
block 105261: Completed Completed Completed
block 105451: Completed Completed Completed
block 105641: Completed Completed Completed
block 105831: Completed Completed Completed
block 106021: Completed Completed Completed
block 106211: Completed Completed Completed
block 106401: Completed Completed Completed
block 106591: Completed Completed Completed
block 106781: Completed Completed Completed
block 106971: Completed Completed Completed
block 107161: Completed Completed Completed
block 107351: Completed Completed Completed
block 107541: Completed Completed Completed
block 107731: Completed Completed Completed
block 107921: Completed Completed Completed
block 108111: Completed Completed Completed
block 108301: Completed Completed Completed
block 108491: Completed Completed Completed
block 108681: Completed Completed Completed
block 108871: Completed Completed Completed
block 109061: Completed Completed Completed
block 109251: Completed Completed Completed
block 109441: Completed Completed Completed
block 109631: Completed Completed Completed
block 109821: Completed Completed Completed
block 110011: Completed Completed Completed
block 110201: Completed Completed Completed
block 110391: Completed Completed Completed
block 110581: Completed Completed Completed
block 110771: Completed Completed Completed
block 110961: Completed Completed Completed
block 111151: Completed Completed Completed
block 111341: Completed Completed Completed
block 111531: Completed Completed Completed
block 111721: Completed Completed Completed
block 111911: Completed Completed Completed
block 112101: Completed Completed Completed
block 112291: Completed Completed Completed
block 112481: Completed Completed Completed
block 112671: Completed Completed Completed
block 112861: Completed Completed Completed
block 113051: Completed Completed Completed
block 113241: Completed Completed Completed
block 113431: Completed Completed Completed
block 113621: Completed Completed Completed
block 113811: Completed Completed Completed
block 114001: Completed Completed Completed
block 114191: Completed Completed Completed
block 114381: Completed Completed Completed
block 114571: Completed Completed Completed
block 114761: Completed Completed Completed
block 114951: Completed Completed Completed
block 115141: Completed Completed Completed
block 115331: Completed Completed Completed
block 115521: Completed Completed Completed
block 115711: Completed Completed Completed
block 115901: Completed Completed Completed
block 116091: Completed Completed Completed
block 116281: Completed Completed Completed
block 116471: Completed Completed Completed
block 116661: Completed Completed Completed
block 116851: Completed Completed Completed
block 117041: Completed Completed Completed
block 117231: Completed Completed Completed
block 117421: Completed Completed Completed
block 117611: Completed Completed Completed
block 117801: Completed Completed Completed
block 117991: Completed Completed Completed
block 118181: Completed Completed Completed
block 118371: Completed Completed Completed
block 118561: Completed Completed Completed
block 118751: Completed Completed Completed
block 118941: Completed Completed Completed
block 119131: Completed Completed Completed
block 119321: Completed Completed Completed
block 119511: Completed Completed Completed
block 119701: Completed Completed Completed
block 119891: Completed Completed Completed
block 120081: Completed Completed Completed
block 120271: Completed Completed Completed
block 120461: Completed Completed Completed
block 120651: Completed Completed Completed
block 120841: Completed Completed Completed
block 121031: Completed Completed Completed
block 121221: Completed Completed Completed
block 121411: Completed Completed Completed
block 121601: Completed Completed Completed
block 121791: Completed Completed Completed
block 121981: Completed Completed Completed
block 122171: Completed Completed Completed
block 122361: Completed Completed Completed
block 122551: Completed Completed Completed
block 122741: Completed Completed Completed
block 122931: Completed Completed Completed
block 123121: Completed Completed Completed
block 123311: Completed Completed Completed
block 123501: Completed Completed Completed
block 123691: Completed Completed Completed
block 123881: Completed Completed Completed
block 124071: Completed Completed Completed
block 124261: Completed Completed Completed
block 124451: Completed Completed Completed
block 124641: Completed Completed Completed
block 124831: Completed Completed Completed
block 125021: Completed Completed Completed
block 125211: Completed Completed Completed
block 125401: Completed Completed Completed
block 125591: Completed Completed Completed
block 125781: Completed Completed Completed
block 125971: Completed Completed Completed
block 126161: Completed Completed Completed
block 126351: Completed Completed Completed
block 126541: Completed Completed Completed
block 126731: Completed Completed Completed
block 126921: Completed Completed Completed
block 127111: Completed Completed Completed
block 127301: Completed Completed Completed
block 127491: Completed Completed Completed
block 127681: Completed Completed Completed
block 127871: Completed Completed Completed
block 128061: Completed Completed Completed
block 128251: Completed Completed Completed
block 128441: Completed Completed Completed
block 128631: Completed Completed Completed
block 128821: Completed Completed Completed
block 129011: Completed Completed Completed
block 129201: Completed Completed Completed
block 129391: Completed Completed Completed
block 129581: Completed Completed Completed
block 129771: Completed Completed Completed
block 129961: Completed Completed Completed
block 130151: Completed Completed Completed
block 130341: Completed Completed Completed
block 130531: Completed Completed Completed
block 130721: Completed Completed Completed
block 130911: Completed Completed Completed
block 131101: Completed Completed Completed
block 131291: Completed Completed Completed
block 131481: Completed Completed Completed
block 131671: Completed Completed Completed
block 131861: Completed Completed Completed
block 132051: Completed Completed Completed
block 132241: Completed Completed Completed
block 132431: Completed Completed Completed
block 132621: Completed Completed Completed
block 132811: Completed Completed Completed
block 133001: Completed Completed Completed
block 133191: Completed Completed Completed
block 133381: Completed Completed Completed
block 133571: Completed Completed Completed
block 133761: Completed Completed Completed
block 133951: Completed Completed Completed
block 134141: Completed Completed Completed
block 134331: Completed Completed Completed
block 134521: Completed Completed Completed
block 134711: Completed Completed Completed
block 134901: Completed Completed Completed
block 135091: Completed Completed Completed
block 135281: Completed Completed Completed
block 135471: Completed Completed Completed
block 135661: Completed Completed Completed
block 135851: Completed Completed Completed
block 136041: Completed Completed Completed
block 136231: Completed Completed Completed
block 136421: Completed Completed Completed
block 136611: Completed Completed Completed
block 136801: Completed Completed Completed
block 136991: Completed Completed Completed
block 137181: Completed Completed Completed
block 137371: Completed Completed Completed
block 137561: Completed Completed Completed
block 137751: Completed Completed Completed
block 137941: Completed Completed Completed
block 138131: Completed Completed Completed
block 138321: Completed Completed Completed
block 138511: Completed Completed Completed
block 138701: Completed Completed Completed
block 138891: Completed Completed Completed
block 139081: Completed Completed Completed
block 139271: Completed Completed Completed
block 139461: Completed Completed Completed
block 139651: Completed Completed Completed
block 139841: Completed Completed Completed
block 140031: Completed Completed Completed
block 140221: Completed Completed Completed
block 140411: Completed Completed Completed
block 140601: Completed Completed Completed
block 140791: Completed Completed Completed
block 140981: Completed Completed Completed
block 141171: Completed Completed Completed
block 141361: Completed Completed Completed
block 141551: Completed Completed Completed
block 141741: Completed Completed Completed
block 141931: Completed Completed Completed
block 142121: Completed Completed Completed
block 142311: Completed Completed Completed
block 142501: Completed Completed Completed
block 142691: Completed Completed Completed
block 142881: Completed Completed Completed
block 143071: Completed Completed Completed
block 143261: Completed Completed Completed
block 143451: Completed Completed Completed
block 143641: Completed Completed Completed
block 143831: Completed Completed Completed
block 144021: Completed Completed Completed
block 144211: Completed Completed Completed
block 144401: Completed Completed Completed
block 144591: Completed Completed Completed
block 144781: Completed Completed Completed
block 144971: Completed Completed Completed
block 145161: Completed Completed Completed
block 145351: Completed Completed Completed
block 145541: Completed Completed Completed
block 145731: Completed Completed Completed
block 145921: Completed Completed Completed
block 146111: Completed Completed Completed
block 146301: Completed Completed Completed
block 146491: Completed Completed Completed
block 146681: Completed Completed Completed
block 146871: Completed Completed Completed
block 147061: Completed Completed Completed
block 147251: Completed Completed Completed
block 147441: Completed Completed Completed
block 147631: Completed Completed Completed
block 147821: Completed Completed Completed
block 148011: Completed Completed Completed
block 148201: Completed Completed Completed
block 148391: Completed Completed Completed
block 148581: Completed Completed Completed
block 148771: Completed Completed Completed
block 148961: Completed Completed Completed
block 149151: Completed Completed Completed
block 149341: Completed Completed Completed
block 149531: Completed Completed Completed
block 149721: Completed Completed Completed
block 149911: Completed Completed Completed
block 150101: Completed Completed Completed
block 150291: Completed Completed Completed
block 150481: Completed Completed Completed
block 150671: Completed Completed Completed
block 150861: Completed Completed Completed
block 151051: Completed Completed Completed
block 151241: Completed Completed Completed
block 151431: Completed Completed Completed
block 151621: Completed Completed Completed
block 151811: Completed Completed Completed
block 152001: Completed Completed Completed
block 152191: Completed Completed Completed
block 152381: Completed Completed Completed
block 152571: Completed Completed Completed
block 152761: Completed Completed Completed
block 152951: Completed Completed Completed
block 153141: Completed Completed Completed
block 153331: Completed Completed Completed
block 153521: Completed Completed Completed
block 153711: Completed Completed Completed
block 153901: Completed Completed Completed
block 154091: Completed Completed Completed
block 154281: Completed Completed Completed
block 154471: Completed Completed Completed
block 154661: Completed Completed Completed
block 154851: Completed Completed Completed
block 155041: Completed Completed Completed
block 155231: Completed Completed Completed
block 155421: Completed Completed Completed
block 155611: Completed Completed Completed
block 155801: Completed Completed Completed
block 155991: Completed Completed Completed
block 156181: Completed Completed Completed
block 156371: Completed Completed Completed
block 156561: Completed Completed Pending
block 156751: Completed Completed Completed
block 156941: Completed Completed Completed
block 157131: Completed Completed Completed
block 157321: Completed Completed Completed
block 157511: Completed Completed Completed
block 157701: Completed Completed Completed
block 157891: Completed Completed Completed
block 158081: Completed Completed Completed
block 158271: Completed Completed Completed
block 158461: Completed Completed Completed
block 158651: Completed Completed Completed
block 158841: Completed Completed Completed
block 159031: Completed Completed Completed
block 159221: Completed Completed Completed
block 159411: Completed Completed Completed
block 159601: Completed Completed Completed
block 159791: Completed Completed Completed
block 159981: Completed Completed Completed
block 160171: Completed Completed Completed
block 160361: Completed Completed Completed
block 160551: Completed Completed Completed
block 160741: Completed Completed Completed
block 160931: Completed Completed Completed
block 161121: Completed Completed Completed
block 161311: Completed Completed Completed
block 161501: Completed Completed Completed
block 161691: Completed Completed Completed
block 161881: Completed Completed Completed
block 162071: Completed Completed Completed
block 162261: Completed Completed Completed
block 162451: Completed Completed Completed
block 162641: Completed Completed Completed
block 162831: Completed Completed Completed
block 163021: Completed Completed Completed
block 163211: Completed Completed Completed
block 163401: Completed Completed Completed
block 163591: Completed Completed Completed
block 163781: Completed Completed Completed
block 163971: Completed Completed Completed
block 164161: Completed Completed Pending
block 164351: Completed Completed Completed
block 164541: Completed Completed Completed
block 164731: Completed Pending Completed
block 164921: Completed Pending Pending
block 165111: Pending Completed Completed
block 165301: Pending Completed Completed
block 165491: Waiting Completed Completed
block 7796835: Waiting Completed Completed
panic: Some of the data has been pending for too long!

goroutine 12743 [running]:
github.com/umbracle/minimal/protocol/ethereum.(*Backend).Deliver(0xc0000f0630, 0x49525ec, 0x6, 0xc000000188, 0x4802e60, 0xc00f051f80, 0x0, 0x0)
	/Users/alexeyakhunov/workspace/src/github.com/umbracle/minimal/protocol/ethereum/backend.go:399 +0xb35
github.com/umbracle/minimal/protocol/ethereum.(*task).Run(0xc004dd68c0, 0xc00022e2d0)
	/Users/alexeyakhunov/workspace/src/github.com/umbracle/minimal/protocol/ethereum/backend.go:699 +0x1f5
github.com/umbracle/minimal/protocol/ethereum.(*Backend).runTask(0xc0000f0630, 0xc004dd68c0)
	/Users/alexeyakhunov/workspace/src/github.com/umbracle/minimal/protocol/ethereum/backend.go:167 +0x2f
created by github.com/umbracle/minimal/protocol/ethereum.(*Backend).runSync
	/Users/alexeyakhunov/workspace/src/github.com/umbracle/minimal/protocol/ethereum/backend.go:249 +0x5d
exit status 2
```

--- a/protocol/ethereum/backend.go
+++ b/protocol/ethereum/backend.go
@@ -352,13 +352,16 @@ func (b *Backend) Dequeue() *Job {
 	return job
 }
 
-func (b *Backend) Deliver(context string, id uint32, data interface{}, err error) {
+func (b *Backend) Deliver(context string, peerID string, id uint32, data interface{}, err error) {
 	b.deliverLock.Lock()
 	defer b.deliverLock.Unlock()
 
 	if err != nil {
 		// TODO, we need to set here the thing that was not deliver as waiting to be dequeued again
-		fmt.Printf("=> Failed to deliver (%d): %v\n", id, err)
+		fmt.Printf("=> Failed to deliver peer %s %s (%d): %v\n", context, peerID, id, err)
+
+		// b.logger.Trace("failed to deliver", context, "peer", peerID, "job", id)
+
 		if err := b.queue.updateFailedElem(id, context); err != nil {
 			fmt.Printf("Could not be updated: %v\n", err)
 		}
@@ -367,7 +370,7 @@ func (b *Backend) Deliver(context string, id uint32, data interface{}, err error
 
 	switch obj := data.(type) {
 	case []*types.Header:
-		b.logger.Trace("deliver headers", "count", len(obj))
+		b.logger.Trace("deliver headers", "peerID", peerID, "job", id, "count", len(obj))
 
 		// fmt.Printf("deliver headers %d: %d\n", id, len(obj))
 		if err := b.queue.deliverHeaders(id, obj); err != nil {
@@ -375,7 +378,7 @@ func (b *Backend) Deliver(context string, id uint32, data interface{}, err error
 		}
 
 	case []*types.Body:
-		b.logger.Trace("deliver bodies", "count", len(obj))
+		b.logger.Trace("deliver bodies", "peerID", peerID, "job", id, "count", len(obj))
 
 		// fmt.Printf("deliver bodies %d: %d\n", id, len(obj))
 		if err := b.queue.deliverBodies(id, obj); err != nil {
@@ -383,7 +386,7 @@ func (b *Backend) Deliver(context string, id uint32, data interface{}, err error
 		}
 
 	case [][]*types.Receipt:
-		b.logger.Trace("deliver receipts", "count", len(obj))
+		b.logger.Trace("deliver receipts", "peerID", peerID, "job", id, "count", len(obj))
 
 		// fmt.Printf("deliver receipts %d: %d\n", id, len(obj))
 		if err := b.queue.deliverReceipts(id, obj); err != nil {
@@ -683,25 +686,27 @@ func (t *task) Run() bool {
 	case *HeadersJob:
 		// fmt.Printf("SYNC HEADERS (%s): %d %d\n", id, job.block, job.count)
 
-		t.backend.logger.Trace("sync headers", "id", id, "from", job.block, "count", job.count)
+		t.backend.logger.Trace("sync headers", "peerID", id, "job", t.job.id, "from", job.block, "count", job.count)
 
 		data, err = conn.RequestHeadersRangeSync(ctx, job.block, job.count)
 		context = "headers"
 	case *BodiesJob:
 		// fmt.Printf("SYNC BODIES (%s): %d\n", id, len(job.hashes))
 
-		t.backend.logger.Trace("sync bodies", "id", id, "num", len(job.hashes))
+		t.backend.logger.Trace("sync bodies", "peerID", id, "job", t.job.id, "num", len(job.hashes))
 
 		data, err = conn.RequestBodiesSync(ctx, job.hash, job.hashes)
 		context = "bodies"
 	case *ReceiptsJob:
 		// fmt.Printf("SYNC RECEIPTS (%s): %d\n", id, len(job.hashes))
 
-		t.backend.logger.Trace("sync receipts", "id", id, "num", len(job.hashes))
+		t.backend.logger.Trace("sync receipts", "peerID", id, "job", t.job.id, "num", len(job.hashes))
 
 		data, err = conn.RequestReceiptsSync(ctx, job.hash, job.hashes)
 		context = "receipts"
 	}
+
+	var failed bool
 
 	// set the error to the context content
 	if ctx.Err() != nil {
@@ -712,12 +717,12 @@ func (t *task) Run() bool {
 	if err != nil {
 		if err.Error() == "session closed" {
 			// remove the peer from the list of peers if the session has been closed
-			return true
+			failed = true
 		}
 	}
 
-	t.backend.Deliver(context, t.job.id, data, err)
-	return false
+	t.backend.Deliver(context, t.peerID, t.job.id, data, err)
+	return failed
 }
 
 func (t *task) Close() {

